### PR TITLE
signing-keys: fix a native rpm pubkey importing issue

### PR DESCRIPTION
**Pull request by yunguowei**
With existing implemetation, if private key is imported to native gpg
keyring, native rpm will not import pubkey, which is not correct.

We should check if the accountable pubkey is imported to native rpm, and
make sure it is imported.

Signed-off-by: Yunguo Wei <yunguo.wei@windriver.com>

### CODE REVIEW

File: recipes-core/meta/signing-keys.bbappend [hunk 1]
```diff
@@ -38,36 +38,69 @@ def check_gpg_key(d, keyid):
...
+def check_rpm_pubkey(d, gpg_name):
+    # retrieve gpg_name's keyid
+    keyid = get_gpg_keyid(d, gpg_name)
+
+    cmd = "%s --root %s --dbpath /var/lib/rpm -q gpg-pubkey |cut -d- -f3 |xargs " % \
+          (d.getVar('STAGING_BINDIR_NATIVE', True) + '/rpm', d.getVar('STAGING_DIR_NATIVE', True))
+
+    status, output = oe.utils.getstatusoutput(cmd)
+    pubkeys = output.split()
+    key_matched = 0
```
Comment (jiazhang0) [inline recipes-core/meta/signing-keys.bbappend]: Remove this line.

File: recipes-core/meta/signing-keys.bbappend [hunk 2]
```diff
@@ -38,36 +38,69 @@ def check_gpg_key(d, keyid):
...
-        if os.path.exists(gpg_pubkey):
-            cmd = "%s --root %s --dbpath /var/lib/rpm --import %s > /dev/null" % \
-                (d.getVar('STAGING_BINDIR_NATIVE', True) + '/rpm', \
-                 d.getVar('STAGING_DIR_NATIVE', True), gpg_pubkey)
-
-            status, output = oe.utils.getstatusoutput(cmd)
-            if status:
-                raise bb.build.FuncFailed('Failed to import gpg pub key (%s) to sysroot, fail reason: %s' % \
+        # check native rpm pubkey
+        if check_rpm_pubkey(d, d.getVar("RPM_GPG_NAME", True)) is False:
```
Comment (jiazhang0) [inline recipes-core/meta/signing-keys.bbappend]: Actually I don't think we should care importing pubkey to rpm db.

There are 2 importing operations: importing RPM-GPG-PRIVKEY-<RPM_GPG_NAME> to the gpg keyring <GPG_PATH>, and importing <RPM_GPG_PUBKEY> to native rpm5 db.

Considering the context of what is happening here, the latter should be placed somewhere not here. In fact, the pubkey imported to rpm db is not necessary until the build process reaches to do_rootfs() for a certain image, right? Thus, shall we move this importing operation to the native smart when building it?

File: recipes-core/meta/signing-keys.bbappend (hunk 1 above)
Comment (yunguowei) [inline recipes-core/meta/signing-keys.bbappend]: This will be removed.

File: recipes-core/meta/signing-keys.bbappend (hunk 2 above)
Comment (yunguowei) [inline recipes-core/meta/signing-keys.bbappend]: Yes, only native smart needs this. But I would like to keep importing key here, as the recipe signing-keys is used to manage all operation for all kinds of keys. If we have multiple places to manage keys, it is not easy for futher development. But I can split importing gpg and rpm into two seperated ones. gpg key should be imported before native gpg populate_sysroot(), while adding the task importing rpm key before native smartpm's populsate_sysroot().

Comment (jiazhang0) [inline recipes-core/meta/signing-keys.bbappend]: That sounds good.

**Pull request closed without merging**